app/services/c100_app/address_lookup_service.rb

```diff
@@ -0,0 +1,64 @@
+module C100App
+  class AddressLookupService
+    ORDNANCE_SURVEY_URL = 'https://api.ordnancesurvey.co.uk/places/v1/addresses/postcode'.freeze
+
+    attr_accessor :postcode, :result
+
+    def initialize(postcode)
+      @postcode = postcode
+      @result = MapAddressLookupResults.call(results)
+    end
+
+    def success?
+      !failure?
+    end
+
+    def errors
+      @errors ||= AddressLookupErrors.new
+    end
+
+    private
+
+    def query_params
+      {
+        key: ENV.fetch('ORDNANACE_SURVEY_API_KEY'),
```

Comment:
In the tests, we must always mock a default test value for this, otherwise it is not possible to run the tests locally (and I imagine same on CI) unless the env variable is declared:

```ruby
Failure/Error: key: ENV.fetch('ORDNANACE_SURVEY_API_KEY'),
     
     KeyError:
       key not found: "ORDNANACE_SURVEY_API_KEY"
     # ./app/services/c100_app/address_lookup_service.rb:24:in `fetch'
```